### PR TITLE
Add support for 4-bit fused-rowwise-quantized SLWS to Interpreter

### PR DESCRIPTION
**Pull request by jfix71**
Summary: This PR adds support for 4-bit rowwise quantization to FRWQ-SLWS.

Test Plan: Added tests to cover this. I also added a new set of tests (see second commit) that has two columns in addition to being weighted -- previously all of our weighted tests only had a single column of data.

Related to https://github.com/pytorch/glow/issues/3463

CC: @jsubag

### CODE REVIEW

File: lib/Graph/Graph.cpp [hunk 1]
```diff
@@ -1680,28 +1680,39 @@ Function::createRowwiseQuantizedSparseLengthsSum(
...
 /// createRowwiseQuantizedSparseLengthsSum and
 /// createRowwiseQuantizedSparseLengthsWeightedSum.
-/// Function \p F is used to get the speficific type, using inputs \p inDims and
-/// \p lenghtsDims to compute output dimensions.
-static TypeRef getOutputTypeOfFusedRowwiseQuantizedSLS(
-    Function *F, const llvm::ArrayRef<size_t> &inDims,
-    const llvm::ArrayRef<size_t> &lengthsDims, ElemKind scaleOffsetKind) {
-  ShapeVector outDims(inDims.begin(), inDims.end());
+/// Function \p F is used to get the specific type, using inputs \p data and
+/// \p lengthsDims to compute output dimensions. \p dataKind represents the kind
```
Comment (beicy) [inline lib/Graph/Graph.cpp]: Where is dataKind?

File: lib/Graph/Graph.cpp [hunk 2]
```diff
@@ -1758,21 +1781,22 @@ static Constant *quantizeDataForFusedRowwiseQuantizedSparseLengthsWeightedSum(
...
+          this, data, fusedElemKind);
   return createFusedRowwiseQuantizedSparseLengthsWeightedSum(
       name, rwqData, weights, indices, lengths, precision, useFP16Accumulation);
 }
 
 FusedRowwiseQuantizedSparseLengthsSumNode *
 Function::createFusedRowwiseQuantizedSparseLengthsSum(
     llvm::StringRef name, Tensor &data, NodeValue indices, NodeValue lengths,
-    ElemKind precision, bool useFP16Accumulation) {
+    ElemKind precision, bool useFP16Accumulation, ElemKind fusedElemKind) {
```
Comment (beicy) [inline lib/Graph/Graph.cpp]: since now there is fusedElemKind, do we still need precision?  I mean, for 4bit, it must be float16Ty.

**Comment by jackm321** [comment]
LGTM!
The pytorch failure is unrelated and should be fixed by #3717

File: lib/Graph/Graph.cpp (hunk 2 above)
Comment (jfix71) [reply lib/Graph/Graph.cpp]: Good point @beicy -- I added a new commit where I went through and cleaned up all the APIs that don't need precision. Now we will pass the fused ElemKind and then infer the precision from the equivalent ElemKind used by scale/offset.

**Pull request closed without merging**